open-craft/labxchange-xblocks
# [LX-2281] Adds BlockstoreService for when openedx is not available

This PR allows these XBlocks to be run outside of Open edX by adding an optional "blockstore" service, which knows how to interact with the Blockstore/Content Libraries APIs to access the data that Open edX access directly.

The BlockstoreService itself is in the LabXchange code, but can be upstreamed if it's found to be broadly useful. (So far, no one else but LabXchange are using these XBlocks.)

Here, we also change the license from AGPL to Apache 2.0, so this repo can be included in the LabXchange private source code.

**Testing instructions**

This branch is installed as part of [labxchange-dev#1632](https://gitlab.com/opencraft/client/LabXchange/labxchange-dev/-/merge_requests/1632), so part of this change can be tested as part of that.

Additionally, please test that this branch works fine in the LMS/Studio by:

* Installing this branch in the lms and studio virtualenvs.
* Viewing audio and video XBlocks in LabXchange, and ensuring you can: watch the videos, read the transcripts, edit the metadata and OLX, and see the changes made.

Automated testing:

1. [Install this branch on devstack](https://github.com/open-craft/labxchange-xblocks/blob/jill/lx-2281/README.rst#installation-on-devstack).
2. [Follow instructions to run tests](https://github.com/open-craft/labxchange-xblocks/blob/jill/lx-2281/README.rst#testing-on-devstack)

**Author notes & concerns**

1. Currently, 7 tests are failing on the `master` branch of this repo.
   These tests are still failing in this branch; I can remedy them if required.
1. After merging, we'll tag a new release: `0.9.0`
